plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/test/block.ts

```diff
@@ -188,149 +188,63 @@ async function switchProductType( productType: string ) {
 	} );
 }
 
+const expectHasBlock = async ( blockName: string ) => {
+	const block = await screen.findAllByLabelText( `Block: ${ blockName }` );
+	expect( block.length ).toBeGreaterThan( 0 );
+};
+
 describe( 'Add to Cart + Options block', () => {
-	it( 'should render inner blocks', async () => {
+	it( 'should render inner blocks for simple and external products', async () => {
 		await setup();
-
-		const block = screen.getByRole( 'document', {
-			name: 'Block: Add to Cart + Options (Beta)',
-		} );
-		expect( block ).toBeInTheDocument();
+		await expectHasBlock( 'Add to Cart + Options (Beta)' );
 
 		// Simple products.
-		await waitFor( () => {
-			expect(
-				screen.getByRole( 'document', {
-					name: 'Block: Product Stock Indicator',
-				} )
-			).toBeInTheDocument();
-
-			expect(
-				screen.getByRole( 'document', {
-					name: 'Block: Product Quantity (Beta)',
-				} )
-			).toBeInTheDocument();
-
-			expect(
-				screen.getByRole( 'document', {
-					name: 'Block: Add to Cart Button',
-				} )
-			).toBeInTheDocument();
-		} );
+		await expectHasBlock( 'Product Stock Indicator' );
+		await expectHasBlock( 'Product Quantity (Beta)' );
+		await expectHasBlock( 'Add to Cart Button' );
 
 		// External products.
 		await switchProductType( 'External/Affiliate product' );
 
 		await waitFor( () => {
 			expect(
-				screen.queryByRole( 'document', {
-					name: 'Block: Product Stock Indicator',
-				} )
+				screen.queryByLabelText( 'Block: Product Stock Indicator' )
 			).not.toBeInTheDocument();
-
-			expect(
-				screen.getByRole( 'document', {
-					name: 'Block: Add to Cart Button',
-				} )
-			).toBeInTheDocument();
 		} );
+		await expectHasBlock( 'Add to Cart Button' );
+	} );
 
-		// Grouped products.
-		await switchProductType( 'Grouped product' );
-
-		await waitFor( () => {
-			expect(
-				screen.queryByRole( 'document', {
-					name: 'Block: Grouped Product Selector (Beta)',
-				} )
-			).toBeInTheDocument();
+	it( 'should render inner blocks for grouped products', async () => {
+		expect.hasAssertions();
```

Comment:
From doc:

> expect.hasAssertions() verifies that at least one assertion is called during a test. This is often useful when testing asynchronous code, in order to make sure that assertions in a callback actually got called.

Does it mean it's enough that one expect in `expectHasBlock` pass for test to pass? 🤔

Reply:
Ah, I see, thanks for explanation! Makes sense!